aiodebug/hang_inspection.py

```diff
@@ -0,0 +1,98 @@
+from typing import List, Optional
+import sys
+import traceback
+import asyncio
+import time
+import os
+import threading
+import datetime
+
+from pygments import highlight
+from pygments.lexers import PythonLexer
+from pygments.formatters import HtmlFormatter
+
+
+
+class TraceDumper(threading.Thread):
+
+	def __init__(self, dir_name: str, interval: float, last_loop_iteration_time_wrapped: List[float], lock: threading.Lock) -> None:
+		self.interval = interval
+		self.dir_name = os.path.abspath(dir_name)
+		self.last_loop_iteration_time_wrapped = last_loop_iteration_time_wrapped
+		self.lock = lock
+		self.stop = False
+		self.monitor_task = None  # type: Optional[asyncio.Task]
+
+		# Eliminate IO errors later (on hang)
+		assert os.path.isdir(self.dir_name)
+		assert os.access(self.dir_name, os.W_OK)
+
+		threading.Thread.__init__(self)
+
+
+	def run(self) -> None:
+		'''Thread loop'''
+		while not self.stop:
+			with self.lock:
```

Comment:
I don't think we need the lock. Lists are thread-safe. http://stackoverflow.com/questions/6319207/are-lists-thread-safe